src/movies.js

```diff
@@ -1,26 +1,84 @@
 // The `movies` array from the file `src/data.js`.
-console.log('movies: ', movies);
+//console.log('movies: ', movies);
+
+const movies = require("./data");
 
 
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(array) {
+  const mappedArray = array.map(function (element){
+    return element.director;
+  });
+    return mappedArray;   
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(array) {
+  const countDramaOnly = array.filter(function (element) {
+    return (element.director === 'Steven Spielberg' && element.genre.includes('Drama'));
+
+  });
+  return countDramaOnly.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(array) {
+  if (array.length === 0) return 0;
+  
+
+  const countScores = array.reduce(function(accumulator, currentValue) {
+    return accumulator + currentValue.score;},0);
```

Comment:
Here you are missing one little step. Some movies in data do not have score. They have property name , but not the value. 

so indide reduce method you want to check if `currentValue.score `exists
  - if it does `return accumulator + currentValue.score`
  - if it does not` return accumulator`